tests/test_remote.py

```diff
@@ -868,3 +872,130 @@ async def test_regeneration_without_warning_for_good_dns_config(
     await asyncio.sleep(0.1)
 
     assert snitun_mock.call_disconnect
+
+
+async def test_certificate_task_renew_cert_jws_failiure(
+    auth_cloud_mock, mock_cognito, aioclient_mock, snitun_mock
+):
+    """Initialize backend."""
+    auth_cloud_mock.servicehandlers_server = "test.local"
+
+    class _MockAcmeClient(client.ClientV2):
+        def __init__(self) -> None:
+            pass
+
+        def new_order(self, _):
+            raise messages.Error.from_json(
+                {
+                    "type": "urn:ietf:params:acme:error:malformed",
+                    "detail": "JWS verification error",
+                }
+            )
+
+    class _MockAcme(AcmeHandler):
```

Comment:
Why couldn't we use the existing `acme_mock`?

Reply:
I guess it's because it's not a MagicMock, but a custom class, so we can't set side effects?

Reply:
Yeah, to properly do this, it's need to be handled deeper then what acme_mock allows

Reply:
I'd probably remove the custom class and use the `wraps` keyword when patching and set the required return values and side effects of the five attributes that we need to adjust. That could make it easier to raise exceptions and assert calls.

Something to consider for the future.

Reply:
For the future I really want to redo all testing in the lib...

Reply:
I know what you mean. :smile: :+1: